cassandra/cassandra_test.go

```diff
@@ -0,0 +1,441 @@
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	cassandracontainer "github.com/testcontainers/testcontainers-go/modules/cassandra"
+)
+
+const (
+	// cassandraImage is the default image used for running cassandra in tests.
+	cassandraImage              = "cassandra:latest"
+	cassandraImageEnvVar string = "TEST_CASSANDRA_IMAGE"
+	cassandraPort               = "9042/tcp"
+)
+
+// newTestStore creates a Cassandra container using the official module
+func newTestStore(t testing.TB) string {
+	t.Helper()
+
+	img := cassandraImage
+	if imgFromEnv := os.Getenv(cassandraImageEnvVar); imgFromEnv != "" {
+		img = imgFromEnv
+	}
+
+	ctx := context.Background()
+
+	c, err := cassandracontainer.Run(ctx, img)
+	testcontainers.CleanupContainer(t, c)
+	require.NoError(t, err)
+
+	// Get connection parameters
+	host, err := c.Host(ctx)
+	require.NoError(t, err)
+
+	mappedPort, err := c.MappedPort(ctx, cassandraPort)
+	require.NoError(t, err)
+
+	connectionURL := host + ":" + mappedPort.Port()
+	return connectionURL
+}
+
+// TestCassandraStorage tests the Cassandra storage implementation
+func TestCassandraStorage(t *testing.T) {
+	// Test cases
+	t.Run("KeyspaceCreation", func(t *testing.T) {
```

Comment:
suggestion: even though I like the subtest pattern, I'd recommend following the same test structure as in the other storages, using separate test functions with the same names.

I discovered this pattern while working more and more in the test code base, and it allowed me creating a mental model around what is being tested for each storage. Even more, I'm thinking about proposing an abstraction so that all storages use the same code, something like a TCK: a test compatibility kit that each storage must satisfy.

For that reason, I'd try to have the same test functions plus those specific to the Cassandra storage, if any.